app/config.go

```diff
@@ -681,14 +681,30 @@ func createDummyACL(kp crypto.PublicKey) govTypes.ACL {
 // XXX: this is totally unsafe.
 // it's only suitable for testnets.
 func ResetWorldState(cmd *cobra.Command, args []string) {
+	loadGlobaConfigForReset(cmd)
 	ResetAll(GlobalConfig.TendermintConfig.DBDir(),
 		GlobalConfig.TendermintConfig.P2P.AddrBookFile(),
 		GlobalConfig.PocketConfig.DataDir+FS+GlobalConfig.TendermintConfig.PrivValidatorKey,
 		GlobalConfig.PocketConfig.DataDir+FS+GlobalConfig.TendermintConfig.PrivValidatorState,
 		log.NewNopLogger())
 }
 
-// ResetAll removes address book files plus all data, and resets the privValdiator data.
+func loadGlobaConfigForReset(cmd *cobra.Command) {
+	var datadir string
+	if cmd.Flag("datadir").DefValue == cmd.Flag("datadir").Value.String() {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			panic("could not get home directory for data dir creation: " + err.Error())
+		}
+		datadir = home + FS + DefaultDDName
+	} else {
+		datadir = cmd.Flag("datadir").Value.String()
+	}
+	c := DefaultConfig(datadir)
+	GlobalConfig = c
+}
```

Comment:
This looks very specific to the reset CLI, can we just put this inside the Reset WorldState ?

Reply:
sure